src/values.ts

```diff
@@ -1,21 +1,166 @@
 /**
- * An Instant represents a fixed point in time (called "exact time"), without
- * regard to calendar or location, e.g. July 20, 1969, at 20:17 UTC.
+ * An wrapper around the Fauna `Time` type. It, represents a fixed point in time
+ * without regard to calendar or location, e.g. July 20, 1969, at 20:17 UTC.
  *
- * @remarks This class follows the TC39 proposal for `Temporal.Instant`. See the
- * proposal here: https://tc39.es/proposal-temporal/docs/index.html.
- * Since fauna has both `Time` and `Date` data types, we want to make a clear
- * distinction between them and avoid using the native `Date` class.
+ * @remarks The Javascript `Date` type most closely resembles a Fauna `Time`,
+ * not a Fauna `Date`. However, Fauna stores `Time` values with nanosecond
+ * precision, while Javascript `Date` values only have millisecond precision.
+ * This FaunaTime class preserves precision by storing the original string value
+ * and should be used whenever possible to pass `Time` values back to Fauna.
+ * Converting to a Javascript date before sending to Fauna could result in loss
+ * of precision.
+ *
+ * TODO: link to beta docs for `Time` type
  */
-class FaunaTime {}
+export class FaunaTime {
+  readonly #value: string;
+
+  /**
+   * @remarks constructor is private to enforce using factory functions
+   */
+  private constructor(isoString: string) {
+    this.#value = isoString;
+  }
+
+  /**
+   * Creates a new {@link FaunaTime} from an ISO date string
+   * @param item - An ISO date string.
+   * @returns A new {@link FaunaTime}
+   * @throws TypeError if a string is not provided, or RangeError if item
+   * is not a valid date
+   */
+  static from(item: string): FaunaTime {
+    if (typeof item !== "string") {
+      throw new TypeError(
+        `Expected string but received ${typeof item}: ${item}`
+      );
+    }
+    const validatedDate = new Date(item);
+    if (validatedDate.toString() === "Invalid Date") {
+      throw new RangeError(
+        `Expected a valid date string but received '${item}'`
+      );
+    }
+
+    return new FaunaTime(item);
+  }
+
+  /**
+   * Creates a new {@link FaunaTime} from a Javascript `Date`
+   * @param isoString - A Javascript `Date`
+   * @returns A new {@link FaunaTime}
+   */
+  static fromDate(date: Date): FaunaTime {
+    return new FaunaTime(date.toISOString());
+  }
+
+  /**
+   * Get the underlying string value
+   */
+  get value(): string {
+    return this.#value;
+  }
```

Comment:
I would call this isoString

makes it clear to users what it is.

value is a bit ambiguous